### PR TITLE
Fix for small resolutions

### PR DESCRIPTION
**Pull request by ShevchenkoVtwo**
- Add gradient to mask
- Fix UI for small resolutions
- In ZIP two video's of app work on different resolutions 
[Homework demonstration.zip](https://github.com/ShevchenkoVtwo/HomeworkAndroidAcademy/files/5573849/Homework.demonstration.zip)

### CODE REVIEW

File: app/src/main/res/layout-land/activity_movie_detail.xml [hunk 1]
```diff
@@ -183,111 +183,27 @@
                 app:layout_constraintStart_toStartOf="parent"
                 app:layout_constraintTop_toBottomOf="@+id/txMovieDescription" />
 
-            <ImageView
-                android:id="@+id/imActor1"
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
+            <HorizontalScrollView
```
Comment (emartynov) [inline app/src/main/res/layout-land/activity_movie_detail.xml]: The less layout files you have - the better. Great that you extracted common part to the separate layout. But landscape layout is redundant for this app (at least for now).

File: app/src/main/res/layout-land/activity_movie_detail.xml [hunk 2]
```diff
@@ -183,111 +183,27 @@
...
-                android:id="@+id/txActor2"
-                android:layout_width="80dp"
-                android:layout_height="wrap_content"
-                android:layout_marginTop="8dp"
-                android:layout_marginBottom="8dp"
-                android:fontFamily="@font/roboto_medium"
-                android:text="Chris
-                Evans"
-                android:textColor="@color/title_color"
+                android:layout_marginEnd="16dp"
```
Comment (emartynov) [inline app/src/main/res/layout-land/activity_movie_detail.xml]: Use dimens.xml file for such constants

File: app/src/main/res/layout-land/activity_movie_detail.xml [hunk 3]
```diff
@@ -183,111 +183,27 @@
...
-                app:layout_constraintEnd_toEndOf="@+id/imActor4"
-                app:layout_constraintStart_toStartOf="@+id/imActor4"
-                app:layout_constraintTop_toBottomOf="@+id/imActor4"
-                app:layout_constraintVertical_bias="0.0" />
         </androidx.constraintlayout.widget.ConstraintLayout>
+
     </ScrollView>
 
-</androidx.constraintlayout.widget.ConstraintLayout>
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (emartynov) [inline app/src/main/res/layout-land/activity_movie_detail.xml]: Wow, constraint layout over scroll layout is something probably wrong in 90% cases.

File: app/src/main/res/layout/view_actors.xml [hunk 4]
```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:background="@color/movie_background">
+
+    <ImageView
+        android:id="@+id/imActor1"
```
Comment (emartynov) [inline app/src/main/res/layout/view_actors.xml]: Лучше не использовать абревиатуры. И общий стиль, camel_case для ресурсов.

**Review by emartynov — COMMENTED** [review]
Switched to Russian at some point - apologies!

The constraint layout over the scroll layout is really redundant.

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 5]
```diff
@@ -1,217 +1,201 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
```
Comment (ShevchenkoVtwo) [inline app/src/main/res/layout/activity_movie_detail.xml]: Deleted unnecessary constarintLayout

File: app/src/main/res/layout/view_actors.xml [hunk 6]
```diff
@@ -6,89 +6,89 @@
     android:background="@color/movie_background">
 
     <ImageView
-        android:id="@+id/imActor1"
+        android:id="@+id/im_robert"
```
Comment (ShevchenkoVtwo) [inline app/src/main/res/layout/view_actors.xml]: Rename with camel_case

File: app/src/main/res/values/dimens.xml [hunk 7]
```diff
@@ -1,3 +1,8 @@
 <resources>
-    <dimen name="fab_margin">16dp</dimen>
+    <dimen name="actors_margin">8dp</dimen>
+    <dimen name="views_margin_16">16dp</dimen>
+    <dimen name="views_margin_8">8dp</dimen>
+    <dimen name="text_storyline_padding">4dp</dimen>
+    <dimen name="text_description_padding">20dp</dimen>
+    <dimen name="text_cast_padding">8dp</dimen>
```
Comment (ShevchenkoVtwo) [inline app/src/main/res/values/dimens.xml]: Add dimens

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 8]
```diff
@@ -1,217 +1,201 @@
...
-                    android:layout_width="wrap_content"
-                    android:layout_height="match_parent"
-                    android:isScrollContainer="false" />
-            </HorizontalScrollView>
-
-        </androidx.constraintlayout.widget.ConstraintLayout>
-
-    </ScrollView>
-
-</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (ShevchenkoVtwo) [inline app/src/main/res/layout/activity_movie_detail.xml]: Delete unnecessary ConstraintLayout

File: app/src/main/res/layout/view_actors.xml (hunk 6 above)
Comment (emartynov) [reply app/src/main/res/layout/view_actors.xml]: What if I decide to put another actor in the first place?

File: app/src/main/res/values/dimens.xml (hunk 7 above)
Comment (emartynov) [reply app/src/main/res/values/dimens.xml]: You already have a couple of 8dp duplications. If you go like this the file will grow enormously.

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 9]
```diff
@@ -1,284 +1,201 @@
...
+            android:layout_marginStart="@dimen/views_margin_8"
+            android:fontFamily="@font/roboto"
+            android:text="@string/image_back"
+            android:textColor="@color/arrow_color"
+            android:textSize="12sp"
+            app:layout_constraintBottom_toBottomOf="@+id/im_back"
+            app:layout_constraintStart_toEndOf="@+id/im_back"
+            app:layout_constraintTop_toTopOf="@+id/im_back" />
 
         <ImageView
```
Comment (emartynov) [inline app/src/main/res/layout/activity_movie_detail.xml]: Read about compond drawable

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 10]
```diff
@@ -1,284 +1,201 @@
...
+            app:layout_constraintTop_toTopOf="parent">
+
+            <ImageView
+                android:id="@+id/im_title"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:scaleType="fitXY"
+                app:srcCompat="@drawable/mask" />
+
+            <View
```
Comment (emartynov) [inline app/src/main/res/layout/activity_movie_detail.xml]: Check ImageView foreground

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 11]
```diff
@@ -9,52 +10,32 @@
...
-                app:srcCompat="@drawable/mask" />
-
-            <View
-                android:layout_width="match_parent"
-                android:layout_height="match_parent"
-                android:background="@drawable/gradient" />
-
-        </FrameLayout>
+            app:layout_constraintTop_toTopOf="parent"
+            app:srcCompat="@drawable/gradient" />
```
Comment (ShevchenkoVtwo) [inline app/src/main/res/layout/activity_movie_detail.xml]: Remove unnecessary Frame Layout and fix gradient

File: app/src/main/res/layout/view_actors.xml [hunk 12]
```diff
@@ -6,89 +6,89 @@
     android:background="@color/movie_background">
 
     <ImageView
-        android:id="@+id/im_robert"
+        android:id="@+id/im_first_actor"
```
Comment (ShevchenkoVtwo) [inline app/src/main/res/layout/view_actors.xml]: Rename imageViews and TextViews

File: app/src/main/res/values/dimens.xml [hunk 13]
```diff
@@ -1,5 +1,4 @@
 <resources>
-    <dimen name="actors_margin">8dp</dimen>
     <dimen name="views_margin_16">16dp</dimen>
```
Comment (ShevchenkoVtwo) [inline app/src/main/res/values/dimens.xml]: Delete unnecessary margin

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 14]
```diff
@@ -9,52 +10,32 @@
...
-
-        <ImageView
-            android:id="@+id/im_back"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
-            android:layout_marginStart="@dimen/views_margin_16"
-            android:layout_marginTop="60dp"
-            android:scaleType="fitXY"
+            app:drawableStartCompat="@drawable/ic_arrow_back"
             app:layout_constraintStart_toStartOf="parent"
```
Comment (ShevchenkoVtwo) [inline app/src/main/res/layout/activity_movie_detail.xml]: Add compound drawable

File: app/src/main/res/layout/view_actors.xml [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+        app:layout_constraintStart_toStartOf="@+id/im_second_actor"
+        app:layout_constraintTop_toBottomOf="@+id/im_second_actor" />
+
+    <TextView
+        android:id="@+id/tx_third_actor"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/views_margin_8"
+        android:fontFamily="@font/roboto_medium"
+        android:text="Mark
```
Comment (emartynov) [inline app/src/main/res/layout/view_actors.xml]: Extract to string resources please

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 16]
```diff
@@ -1,284 +1,182 @@
...
+            android:text="@string/cast"
+            android:textColor="@color/title_color"
+            android:textStyle="bold"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/tx_movie_description" />
+
+        <HorizontalScrollView
             android:layout_width="match_parent"
             android:layout_height="match_parent"
```
Comment (emartynov) [inline app/src/main/res/layout/activity_movie_detail.xml]: wrap_content?

File: app/src/main/res/layout/activity_movie_detail.xml [hunk 17]
```diff
@@ -1,284 +1,182 @@
...
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
 
         <ImageView
-            android:id="@+id/imTitle"
-            android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            android:layout_marginBottom="24dp"
+            android:id="@+id/im_title"
```
Comment (emartynov) [inline app/src/main/res/layout/activity_movie_detail.xml:13]: Just question - why do you use `Hungarian` notation for ids?

**Review by emartynov — APPROVED** [review]
Two minor comments left. Up to you if you want to fix them.

File: app/src/main/res/layout/activity_movie_detail.xml (hunk 17 above)
Comment (ShevchenkoVtwo) [reply app/src/main/res/layout/activity_movie_detail.xml:13]: For me it's more understandable when I use id in code

File: app/src/main/res/layout/view_actors.xml (hunk 15 above)
Comment (ShevchenkoVtwo) [reply app/src/main/res/layout/view_actors.xml]: Done

File: app/src/main/res/layout/activity_movie_detail.xml (hunk 16 above)
Comment (ShevchenkoVtwo) [reply app/src/main/res/layout/activity_movie_detail.xml]: Fix

**Pull request merged**